clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4152,6 +4152,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (SYCLStdArg) {
       SYCLStdArg->render(Args, CmdArgs);
       CmdArgs.push_back("-fsycl-std-layout-kernel-params");
+
+      unsigned version = llvm::StringSwitch<unsigned>(SYCLStdArg->getValue())
+                             .Cases("2017", "1.2.1", "121", "sycl-1.2.1", 2017)
+                             .Case("2020", 2020)
+                             .Default(0);
```

Comment:
As you moved the preprocessor settings, this is unused - needs cleanup?

Reply:
good catch. done.